.github/workflows/macos-ci.yml

```diff
@@ -21,49 +21,27 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          #- macos-11
-          - macos-12
+          - macos-13
+          - macos-14
+            # M1 processor
         compiler:
           - clang
           - gcc
-        homebrew-gl:
-          - true
-          # - false
-        homebrew-al:
-          - true
-          - false
 
     steps:
+      # specific setup for Python on Mac to ensure Python Version align
+      - name: Python Setup on Mac
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: 3.12.8
```

Comment:
@stephengtuggy thanks - that greatly helped out; now it's complaining about the `encodings` Python package so a different issue. That really helped.

Reply:
@BenjamenMeyer No problem.

It looks like what's happening now is that the system Xcode installation of Python is taking precedence over what we've installed. Are you setting `CMAKE_FIND_FRAMEWORK ` to `"LAST"` anywhere? I don't see it. Setting that would probably help with this issue.

Reply:
@stephengtuggy thanks - no, I had not. Added that in the lastest version and swapping back to the more specific 3.12.8 for all Mac builds.

Reply:
@stephengtuggy thanks! after requiring C++14 and setting that we now have MacOS 13 support again. MacOS 14+ is on M1+ ARM processors and there seems to be some linker issues. I'm happy with this PR just getting MacOS 13 going again. I might try RPi for ARM and see what happens; though it could be brew support too. Not sure, and not going to dive in right now. I'll leave it for someone more expert on Mac and with a Mac M1+ to figure out. Hopefully that'll happen before MacOS 13 gets dropped by GitHub, otherwise we'll have to drop Mac support.